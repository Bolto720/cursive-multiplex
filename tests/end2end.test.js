const { setup } = require('shellshot');

setup();

async function cargo_e2e(name) {
    await expect.command(`cargo build --bin end2end_${name}`)
        .forExitCode(exp => exp.toBe(0));
    await expect.command(
        `tmux new-session -x 80 -y 24 -d ../target/debug/end2end_${name} \; set status off && sleep 0.05`
    ).forExitCode(exp => exp.toBe(0));
    await expect.command('tmux capture-pane -J -p -t %0')
        .forStdout(exp => exp.toMatchSnapshot());
    await expect.command('tmux kill-server')
        .forExitCode(exp => exp.toBe(0));
}

it('runs a basic smoke test', async () => await cargo_e2e('smoke'));
it('tests the horizontal splitting', async () => await cargo_e2e('horizontal'));
it('tests the vertical splitting', async () => await cargo_e2e('vertical'));
it('tests the horizontal layout with fixed size child',
   async () => await cargo_e2e('horizontal_fixed_size'));
it('tests the vertical layout with fixed size child',
   async () => await cargo_e2e('vertical_fixed_size'));
it('tests a complex pane setup', async () => await cargo_e2e('complex'));
it('tests switching panes in a horizontal setup',
   async () => await cargo_e2e('horizontal_switch_views'));
it('tests switching panes in a vertical setup',
   async () => await cargo_e2e('vertical_switch_views'));
it('tests switching panes in a complex setup',
   async () => await cargo_e2e('complex_switch_views'));
it('tests removing panes in a horizontal setup',
   async () => await cargo_e2e('horizontal_remove'));
it('tests removing panes in a vertical setup',
   async () => await cargo_e2e('vertical_remove'));
it('tests removing panes in a complex setup',
   async () => await cargo_e2e('complex_remove'));
