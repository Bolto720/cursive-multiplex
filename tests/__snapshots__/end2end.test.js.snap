// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs a basic smoke test 1`] = `
"                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                   Hello World                                                  
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
"
`;

exports[`tests a complex pane setup 1`] = `
"left 1                                  │right 1            │right 2            
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
────────────────────────────────────────│                   │                   
left 3                                  │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
────────────────────────────────────────│───────────────────────────────────────
left 2                                  │right 3                                
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
"
`;

exports[`tests removing panes in a complex setup 1`] = `
"left 1                                  │right 1            │right 2            
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
────────────────────────────────────────│                   │                   
left 3                                  │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
────────────────────────────────────────│                   │                   
left 2                                  │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
"
`;

exports[`tests removing panes in a horizontal setup 1`] = `
"Left                                    │Right                                  
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
"
`;

exports[`tests removing panes in a vertical setup 1`] = `
"Up                                                                              
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
────────────────────────────────────────────────────────────────────────────────
Down                                                                            
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
"
`;

exports[`tests switching panes in a complex setup 1`] = `
"left 1                                  │left 3             │right 2            
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
────────────────────────────────────────│                   │                   
right 1                                 │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
                                        │                   │                   
────────────────────────────────────────│───────────────────────────────────────
left 2                                  │right 3                                
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
"
`;

exports[`tests switching panes in a horizontal setup 1`] = `
"Left                                    │Right                                  
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
                                        │                                       
"
`;

exports[`tests switching panes in a vertical setup 1`] = `
"Up                                                                              
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
────────────────────────────────────────────────────────────────────────────────
Down                                                                            
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
"
`;

exports[`tests the horizontal layout with fixed size child 1`] = `
"Root                                    │┌─────────────────────────────────────┐
                                        ││Fixed                                │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        ││                                     │
                                        │└─────────────────────────────────────┘
"
`;

exports[`tests the horizontal splitting 1`] = `
"Right               │Center             │Right                                  
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
                    │                   │                                       
"
`;

exports[`tests the vertical layout with fixed size child 1`] = `
"Root                                                                            
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
────────────────────────────────────────────────────────────────────────────────
┌──────────────────────────────────────────────────────────────────────────────┐
│Fixed                                                                         │
│                                                                              │
│                                                                              │
│                                                                              │
│                                                                              │
│                                                                              │
│                                                                              │
│                                                                              │
│                                                                              │
└──────────────────────────────────────────────────────────────────────────────┘
"
`;

exports[`tests the vertical splitting 1`] = `
"Up                                                                              
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
────────────────────────────────────────────────────────────────────────────────
Center                                                                          
                                                                                
                                                                                
                                                                                
                                                                                
────────────────────────────────────────────────────────────────────────────────
Down                                                                            
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
"
`;
